src/main/java/com/uber/profiling/AgentImpl.java

```diff
@@ -64,7 +64,16 @@ public void run(Arguments arguments, Instrumentation instrumentation, Collection
 
         String processUuid = UUID.randomUUID().toString();
 
-        String appId = SparkUtils.probeAppId(arguments.getAppIdRegex());
+        String appId = null;
+        
+        String appIdVariable = arguments.getAppIdVariable();
+        if (appIdVariable != null && !appIdVariable.isEmpty()) {
+            appId = System.getenv(appIdVariable);
+        }
+        
+        if (appId == null || appId.isEmpty()) {
```

Comment:
generally, we should be consistent with blank string, ie. "  "
however, String class in java doesn't have that built in - consider using apache common StringUtils.isBlank https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html